### PR TITLE
Feat：EXPORT协议

### PR DESCRIPTION
**Pull request by 1Telescope1**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/xun082/create-neat/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Test related changes
- [ ] Security fix
- [ ] Dependency update
- [ ] Revert changes
- [ ] Other... Please describe:

## What is the current behavior?

- Closes #
- Related to #

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/core/src/utils/commonUtils.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import path from "path";
+
+import { FileData } from "../models/FileTree";
+
+/**
+ * 根据给定的文件路径，从嵌套的文件结构中检索目标文件的数据。
+ *
```
Comment (uaenaTzx) [inline packages/core/src/utils/commonUtils.ts:7]: jsdoc 不需要这一行

File: packages/core/src/utils/ast/commonAst.ts [hunk 2]
```diff
@@ -52,3 +52,14 @@ export const createCallExpression = (
...
 export const createObjectProperty = (name: string, value: Parameters<typeof objectProperty>[1]) =>
   objectProperty(identifier(name), value);
+
+/**
+ * 封装遍历ast中ExportDefaultDeclaration的通用函数
+ * @param {object} path
+ * @param {object} t
+ * @param {string} content 要包裹导出的字符串
+ */
+export const exportDefaultDeclarationUtils = (path: any, t: any, content: string) => {
```
Comment (uaenaTzx) [inline packages/core/src/utils/ast/commonAst.ts:62]: any 尽量避免吧

File: packages/core/src/configs/protocol.ts [hunk 3]
```diff
@@ -1,27 +1,22 @@
-const globalProtocol = {
+export const globalProtocol = {
   ENTRY_FILE: "ENTRY_FILE", // 入口文件配置，如：入口文件引入全局 less、scss
+  UPDATE_EXPORT_CONTENT_PROTOCOL: "UPDATE_EXPORT_CONTENT_PROTOCOL",
```
Comment (uaenaTzx) [inline packages/core/src/configs/protocol.ts:3]: 没有 get 到 UPDATE 的语义，INSERT_EXPORT 感觉合适一些

File: packages/core/src/configs/protocol.ts [hunk 4]
```diff
@@ -1,27 +1,22 @@
...
 
 // 插件对构建工具的协议
-const pluginToBuildToolProtocol = {
+export const pluginToBuildToolProtocol = {
   ...globalProtocol,
 };
 
 // 框架对构建工具的协议
-const templateToBuildToolProtocol = {
+export const templateToBuildToolProtocol = {
```
Comment (uaenaTzx) [inline packages/core/src/configs/protocol.ts:19]: 统一导出在性能上会更好

File: packages/core/src/models/protocolGenerator/ProtocolGeneratorAPI.ts [hunk 5]
```diff
@@ -11,7 +13,7 @@ import { Protocol } from "../../types/protocol";
 class ProtocolGeneratorAPI {
   protected protocols: Record<string, object>; // todo 类型考虑优化
   protected props: ProtocolProps;
-  protected protocol: Protocol;
+  protected protocol: string;
```
Comment (uaenaTzx) [inline packages/core/src/models/protocolGenerator/ProtocolGeneratorAPI.ts:16]: 为啥不用 Protocol 了

File: packages/core/src/utils/ast/commonAst.ts [hunk 6]
```diff
@@ -52,3 +52,13 @@ export const createCallExpression = (
  */
 export const createObjectProperty = (name: string, value: Parameters<typeof objectProperty>[1]) =>
   objectProperty(identifier(name), value);
+
+/**
+ * 封装遍历ast中ExportDefaultDeclaration的通用函数
```
Comment (uaenaTzx) [inline packages/core/src/utils/ast/commonAst.ts:57]: ExportDefaultDeclaration的通用函数，需要结合上下文才能看懂是什么意思
可以尝试给一个更清晰的说明

Comment (1Telescope1) [reply packages/core/src/utils/ast/commonAst.ts:57]: 重写了注释

File: packages/core/src/models/protocolGenerator/ProtocolGeneratorAPI.ts (hunk 5 above)
Comment (1Telescope1) [reply packages/core/src/models/protocolGenerator/ProtocolGeneratorAPI.ts:16]: Protocol 最初就给错了，protocol它就是协议的名称，就是一个字符串

File: packages/core/src/configs/protocol.ts (hunk 4 above)
Comment (1Telescope1) [reply packages/core/src/configs/protocol.ts:19]: 修改了，因为受到cjs文件引入，最后改成cjs的导出形式

File: packages/core/src/configs/protocol.ts (hunk 3 above)
Comment (1Telescope1) [reply packages/core/src/configs/protocol.ts:3]: 它的作用 将export default App 转化成，export default Observr(App)，所有叫这个没什么不妥的吧

File: packages/core/src/models/protocolGenerator/ProtocolGeneratorAPI.ts (hunk 5 above)
Comment (uaenaTzx) [reply packages/core/src/models/protocolGenerator/ProtocolGeneratorAPI.ts:16]: 后面可以改成 enum

**Review by uaenaTzx — APPROVED** [review]

**Review by uaenaTzx — APPROVED** [review]

**Pull request merged**